### ISSUE TITLE
Sitemap error in Google Webmaster Tools

### ISSUE THREAD

**Issue by jagbarcelo** [OrchardCMS/OrchardCore#7301]
We have a multlanguage site (en-GB & es-ES) and we have enabled the *Sitemaps* feature.

We have given Google Webmater Tools the URL for the sitemap and it says:

**Your Sitemap or Sitemap index file doesn't properly declare the namespace.**

![image](https://user-images.githubusercontent.com/6704109/96337786-d7156680-1089-11eb-80b3-23d5056b9ce6.png)

The generated sitemap.xml file looks like this:

```
<?xml version="1.0" encoding="utf-8"?>
<urlset xmlns:xsi="http://wwww.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.sitemaps.org/schemas/sitemap/0.9 http://www.sitemaps.org/schemas/sitemap/0.9/sitemap.xsd" xmlns:xhtml="http://www.w3.org/TR/xhtml11/xhtml11_schema.html" xmlns="http://www.sitemaps.org/schemas/sitemap/0.9">
  <url>
    <loc>https://www.my-site.com/en/release-notes</loc>
    <xhtml:link rel="alternate" hreflang="en-GB" href="https://www.my-site.com/en/release-notes" />
    <xhtml:link rel="alternate" hreflang="es-ES" href="https://www.my-site.com/es/notas-de-version" />
    <lastmod>2020-06-03T09:39:03+00:00</lastmod>
    <changefreq>daily</changefreq>
    <priority>0.5</priority>
  </url>
  ... more urls here ...
</urlset>

```
We are using OC 1.0.0-rc2-14802.
Regards.

**Comment by jagbarcelo** [comment]
According to Google in  https://support.google.com/webmasters/answer/189077?hl=en the namespace to be declared should be:
`xmlns:xhtml="http://www.w3.org/1999/xhtml"`

So the sitemap should start with:

```
<?xml version="1.0" encoding="UTF-8"?>
<urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9"
  xmlns:xhtml="http://www.w3.org/1999/xhtml">
  <url>
    ...
```

![image](https://user-images.githubusercontent.com/6704109/96353434-62afe700-10cc-11eb-94bb-b3214930e1b1.png)

**Comment by Skrypt** [comment]
I agree with the proposed change. Please provide a PR.

**Comment by jagbarcelo** [comment]
I have fixed it (still pending to do the PR) and it seems to work fine from Google's point of view:

![image](https://user-images.githubusercontent.com/6704109/96368348-aef23a00-1153-11eb-85c3-54679ba84bb5.png)

The only downside of this is that, having removed the extra namespaces and schemaLocation attribute (to match the simpler headers as Google's example), the xml file is seen as raw (not xml formatted) when viewed through a browser:

This behaviour was already described back in 2014, here: https://stackoverflow.com/a/24737911

I'll make the PR later this week, when I had extensively tested it.

**Comment by agriffard** [comment]
Fixed by #7315

**Comment by jagbarcelo** [comment]
Despite the documentation provided by Google, it seems that it not updated, is updated elsewhere, or they simply do not follow their own rules. The fact is that, after 10 days of having posted my sitemap.xml to Google Webmaster Tools and being accepted as valid then, today I have seen this error (again):

![image](https://user-images.githubusercontent.com/6704109/97634521-266a7800-1a36-11eb-804d-a2ff5651a245.png)

That means, it shows the very same error as it was shown initially. Either they have changed something in their algorithm (not likely) or they accepted initially and it took up to 10 days to report about the error (I've been checking it daily and yesterday no errors were shown).

I do not know if I should create another issue or keep on working on this one until it is resolved for good. Anyway, I'lll keep doing changes to my code and whenever it works in the long term, I'll try to make a final PR (after having tested it for more than 10-20 days without errors).
